### PR TITLE
REST base - validations

### PR DESCRIPTION
**Pull request by pjuza**
**REST base - validations**

I introduced validation model for MVC/REST applications. Main idea was to be able validate RPC objects and re-use from [Spring validation model](http://docs.spring.io/spring/docs/current/spring-framework-reference/htmlsingle/#validation-beanvalidation) as much as possible:
- I changed RPC API - _ChangeableRpc_: added _BindingResult_ to specific methods
- added _org.openhubframework.openhub.api.exception.validation.InputValidationException_ (renamed _ValidationIntegrationException_ -> _ValidationException_)
- changed _ChangeableRpc#validate_ - now there are several possibilities how to make validation:
  - add errors to _BindingResult_ errors (e.g. use _ValidationUtils_). If it has errors then _InputValidationException_ will be thrown
  - throw _ValidationException_ itself (e.g. use _Constraints_ for asserts)
  - if exception is thrown and it's not type of _ValidationException_ then original exception will be wrapped by _IllegalDataException_
  - use [JSR-303 Bean Validation API](http://docs.spring.io/spring/docs/current/spring-framework-reference/htmlsingle/#validation-beanvalidation-overview)

Focus on ChangeableRpc, ExceptionRestHandler#onValidationException and DbConfigurationController#update

### CODE REVIEW

File: core-api/src/main/java/org/openhubframework/openhub/api/common/Constraints.java [hunk 1]
```diff
@@ -154,6 +153,21 @@ public static void isTrue(Boolean expression, String message, ErrorExtEnum error
         }
     }
 
+    /**
+   	 * Assert a boolean expression, throwing {@code IllegalStateException}
+   	 * if the test result is {@code false}. Call isTrue if you wish to
+   	 * throw IllegalArgumentException on an assertion failure.
+   	 * <pre class="code">Assert.state(id == null, "The id property must not already be initialized");</pre>
+   	 * @param expression a boolean expression
```
Comment (hanusto) [inline core-api/src/main/java/org/openhubframework/openhub/api/common/Constraints.java]: probably empty line missing

File: core-api/src/main/java/org/openhubframework/openhub/api/configuration/DbConfigurationParam.java [hunk 2]
```diff
@@ -75,7 +75,6 @@
 
     /** Default public constructor. */
     public DbConfigurationParam() {
```
Comment (hanusto) [inline core-api/src/main/java/org/openhubframework/openhub/api/configuration/DbConfigurationParam.java]: This default constructor can be removed.

File: core-api/src/main/java/org/openhubframework/openhub/api/entity/ExternalCall.java [hunk 3]
```diff
@@ -82,7 +82,6 @@
 
     /** Default public constructor. */
     public ExternalCall() {
```
Comment (hanusto) [inline core-api/src/main/java/org/openhubframework/openhub/api/entity/ExternalCall.java]: Default constructor can be removed.

File: core-api/src/main/java/org/openhubframework/openhub/api/entity/Request.java [hunk 4]
```diff
@@ -80,7 +80,6 @@
 
     /** Default public constructor. */
     public Request() {
```
Comment (hanusto) [inline core-api/src/main/java/org/openhubframework/openhub/api/entity/Request.java]: Default constructor can be removed.

File: core-api/src/main/java/org/openhubframework/openhub/api/entity/Response.java [hunk 5]
```diff
@@ -66,7 +66,6 @@
 
     /** Default public constructor. */
     public Response() {
```
Comment (hanusto) [inline core-api/src/main/java/org/openhubframework/openhub/api/entity/Response.java]: Default constructor can be removed.

File: core-api/src/main/java/org/openhubframework/openhub/api/entity/SuperEntity.java [hunk 6]
```diff
@@ -50,6 +50,13 @@ protected SuperEntity(@Nullable ID id) {
         this.id = id;
     }
 
+    /**
+     * Creates new entity with null identifier.
```
Comment (hanusto) [inline core-api/src/main/java/org/openhubframework/openhub/api/entity/SuperEntity.java]: maybe {@code null} or {@literal null}

**Review by hanusto — APPROVED** [review]
Great improvement. Only small changes are recommended. Maybe one thing - if some validation error occurs and error detail is returned, inputFields structure should be wrapped into another object. It is detail of exception and these information should be grouped together.

**Comment by pjuza** [comment]
Thank you for review, I incorporated your review points ...

**Comment by hanusto** [comment]
I added fix commit to repair failing test - missing constructor on `DbConfigurationParam` entity. It is necessary for Hibernate to instantiate object.

**Pull request merged**